int32 size
ground_msg[] ground_msg_arr